src/refract/SerializeVisitor.cc

```diff
@@ -0,0 +1,128 @@
+//
+//  refract/SerializeVisitor.cc
+//  librefract
+//
+//  Created by Jiri Kratochvil on 18/05/15.
+//  Copyright (c) 2015 Apiary Inc. All rights reserved.
+//
+#include "Element.h"
+
+namespace refract
+{
+
+    void SerializeVisitor::visit(const IElement& e)
+    {
+        result.set("element", sos::String(e.element()));
+
+        typedef IElement::MemberElementCollection::const_iterator iterator;
+
+        // FIXME : refactoring
+        if (!e.meta.empty()) {
+            sos::Object obj;
+            for (iterator it = e.meta.begin(); it != e.meta.end(); ++it) {
+                SerializeCompactVisitor s;
+                s.visit(*(*it));
+                obj.set(s.key(), s.value());
+            }
+            result.set("meta", obj);
+        }
+
+        if (!e.attributes.empty()) {
+            sos::Object obj;
+            for (iterator it = e.attributes.begin(); it != e.attributes.end(); ++it) {
+                SerializeCompactVisitor s;
+                s.visit(*(*it));
+                obj.set(s.key(), s.value());
+            }
+            result.set("attributes", obj);
+        }
+
+        if (e.empty())
+            return;
+
+        if (e.compactContent()) {
+            SerializeCompactVisitor s;
+            e.content(s);
+            result.set("content", s.value());
+        } else {
+            e.content(*this);
+            result.set("content", partial);
+        }
+    }
+
+    void SerializeVisitor::SetSerializerValue(SerializeVisitor& s, sos::Base& value)
+    {
+        if (!s.key.empty()) {
+            s.result.set(s.key, value);
+            s.key.clear();
+        } else {
+            s.partial = value;
+        }
+    }
+
+    void SerializeVisitor::visit(const NullElement& e)
+    {
+        sos::Base value = sos::Null();
+        SetSerializerValue(*this, value);
+    }
+
+    void SerializeVisitor::visit(const StringElement& e)
+    {
+        sos::Base value = sos::Null();
+        if (!e.empty()) {
```

Comment:
IMO, having those empty lines around blocks, makes the code faster and easier to read since our mind gets trained to it.

It is not blocking this PR though.